### PR TITLE
Fixing a hard to trigger bug for `text-generation` pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?
This PR readds sending `attention_mask` to the `generate` function.

In order to trigger the bug ones would need:

- To use the pipeline in `batch_size>1` mode.
- Use a model that did not configure `pad_token_id`. (if it is configured, than generate just recovers the attention mask gracefully)

Then the `generate` function would not be able to recover the attention mask, warn
about it but still generate something (most likely incorrect)

Since the pipeline most likely already generated the `attention_mask` we might as well
send it along to `generate`.

@sgugger 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sugger

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: src/transformers/pipelines/text_generation.py [hunk 1]
```diff
@@ -205,14 +205,18 @@ def preprocess(self, prompt_text, prefix="", handle_long_generation=None, **gene
...
             input_ids = None
+            attention_mask = None
             in_b = 1
         else:
             in_b = input_ids.shape[0]
         prompt_text = model_inputs.pop("prompt_text")
-        generated_sequence = self.model.generate(input_ids=input_ids, **generate_kwargs)  # BS x SL
+        generated_sequence = self.model.generate(
+            input_ids=input_ids, attention_mask=attention_mask, **generate_kwargs
+        )  # BS x SL
```
[the remark this answers is not in the document]
Comment (Narsil) [reply src/transformers/pipelines/text_generation.py:219]: Done !
